providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -0,0 +1,199 @@
+     */
+    public ProviderEvaluation<Integer> integerEvaluation(String key, Integer defaultValue,
+                                                         EvaluationContext ctx) {
+        return resolveGeneric(Integer.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve an object flag
+     */
+    public ProviderEvaluation<Value> objectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        return resolveGeneric(Value.class, key, defaultValue, ctx);
+    }
+
+    private <T> ProviderEvaluation<T> resolveGeneric(Class<T> type, String key, T defaultValue,
+                                                     EvaluationContext ctx) {
+        final FeatureFlag flag = flagStore.getFLag(key);
+
+        // missing flag
+        if (flag == null) {
+            return ProviderEvaluation.<T>builder()
+                    .value(defaultValue)
+                    .reason(Reason.ERROR.toString())
+                    .errorCode(ErrorCode.FLAG_NOT_FOUND)
+                    .errorMessage(String.format("requested flag could not be found: %s", key))
+                    .build();
+        }
+
+        // state check
+        if ("DISABLED".equals(flag.getState())) {
+            return ProviderEvaluation.<T>builder()
+                    .value(defaultValue)
+                    .reason(Reason.DISABLED.toString())
+                    .errorCode(ErrorCode.GENERAL)
+                    .errorMessage(String.format("requested flag is disabled: %s", key))
+                    .build();
+        }
+
+
+        final Object resolvedVariant;
+        final String reason;
+
+        if ("{}".equals(flag.getTargeting())) {
+            resolvedVariant = flag.getDefaultVariant();
+            reason = Reason.STATIC.toString();
+        } else {
+            try {
+                final Object jsonResolved = jsonLogicHandler.apply(flag.getTargeting(), ctx.asObjectMap());
+                if (jsonResolved == null) {
+                    resolvedVariant = flag.getDefaultVariant();
+                    reason = Reason.DEFAULT.toString();
+                } else {
+                    resolvedVariant = jsonResolved;
+                    reason = Reason.TARGETING_MATCH.toString();
+                }
+            } catch (JsonLogicException e) {
+                log.log(Level.FINE, "Error evaluating targeting rule", e);
+                return ProviderEvaluation.<T>builder()
+                        .value(defaultValue)
+                        .reason(Reason.ERROR.toString())
+                        .errorCode(ErrorCode.PARSE_ERROR)
+                        .errorMessage(String.format("error parsing targeting rule: %s", key))
+                        .build();
+            }
+        }
+
+        // check variant existence
+        Object value = flag.getVariants().get(resolvedVariant);
+        if (value == null) {
+            log.log(Level.FINE, String.format("variant %s not found in flag with key %s", resolvedVariant, key));
```

Comment:
```suggestion
            log.log(Level.WARNING, String.format("variant %s not found in flag with key %s", resolvedVariant, key));
```

Reply:
As highlighted at https://github.com/open-feature/java-sdk-contrib/pull/412#discussion_r1316201469 , this could lead to log overflow with incorrect flag configurations/API invocations